mgnify_pipelines_toolkit/utils/rename_contig/cli_rename.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2024-2025 EMBL - European Bioinformatics Institute
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rename command implementation for CLI."""
+
+from pathlib import Path
+
+import click
+
+from .core import read_mapping_file, rename_fasta, write_mapping_file
+from .handlers import GenBankHandler, GFFHandler
+from .parsers import parse_header, parse_virify_header
+
+SUPPORTED_FASTA_EXTS = (".fasta", ".fa", ".fna", ".fasta.gz", ".fa.gz", ".fna.gz")
+SUPPORTED_GBK_EXTS = (".gbk", ".gb", ".genbank")
+SUPPORTED_GFF_EXTS = (".gff", ".gff3", ".gff.gz", ".gff3.gz")
+
+
+@click.command()
+@click.argument("input", nargs=-1, required=True, type=click.Path(exists=True))
+@click.option("-o", "--outdir", required=True, type=click.Path(), help="Output directory")
+@click.option("-p", "--prefix", default="contig", help="Prefix for new names (default: contig)")
+@click.option("-m", "--map", type=click.Path(), help="Output mapping file (TSV)")
+@click.option(
+    "--use-mapping",
+    type=click.Path(exists=True),
+    help="Use existing mapping file instead of generating new names",
```

Comment:
For clarity:
```suggestion
    help="Use input mapping file instead of generating new names",
```

Reply:
Also would be useful to have an example of what the mapping file should look like somewhere... at this stage of reading I have no idea what I should give it if I want to use this parameter